documentation/src/main/docs/config/getting-started.md

```diff
@@ -40,6 +40,26 @@ greeting.message=hello
 goodbye.message=bye
 ```
 
+### Duplicate keys check
+
+If system property :
+
+```
+smallrye.config.forbid-duplicate-keys
+```
+
+Is set to *true* 
+
+A ConfigValidationException will be raised if a duplicate key is found in the same config source.
+
+i.e. 
+
+```
+-Dsmallrye.config.forbid-duplicate-keys=true
```

Comment:
I think we don't need an extra configuration to warn / fail. A simple warning should be enough.

Reply:
Thanks for the feedback, this is yeas on property configuration, I have to think at something for YAML configuration as I was relying on *LoadOptions.isAllowDuplicateKeys()*

Reply:
It ends up using a `Map` in the source, so it can use the same `.put` logic:
https://github.com/smallrye/smallrye-config/blob/1ee672abe1080f3ce212d6f9451f1a3b5a71585f/sources/yaml/src/main/java/io/smallrye/config/source/yaml/YamlConfigSource.java#L66-L98